lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb

```diff
@@ -59,5 +72,17 @@ def build_sas(url)
       @host = matches[0][0]
       @sas_token = matches[0][1]
     end
+
+    def raise_error(e)
+      if e.message && e.message.include?(INVALID_BLOB_EXPIRED_ORIGINAL_MESSAGE)
+        raise INVALID_BLOB_EXPIRED_MESSAGE
+      elsif e.message && e.message.include?(INVALID_BLOB_SIG_WELL_FORMED_MESSAGE)
+        raise INVALID_BLOB_SIG_WELL_FORMED_MESSAGE
+      elsif e.message && e.message.include?(INVALID_BLOB_CONTAINER_MESSAGE)
+        raise INVALID_BLOB_CONTAINER_FORMED_MESSAGE
+      else
+        raise INVALID_BLOB_GENERAL_MESSAGE
```

Comment:
how about the path is not found

Reply:
With  wrong path, the exception was not thrown, and the downloaded file is empty. No errors
So It is not rethrow

Reply:
With the wrong pass, BlobNotFound message will be thrown,
I fixed and rethrow to path not found